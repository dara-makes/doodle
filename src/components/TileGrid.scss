.tile-grid {
  display: grid;
  height: 7em;
  grid-template-columns: repeat(3, 1fr);
  grid-template-rows: auto;
  width: 70%;
}