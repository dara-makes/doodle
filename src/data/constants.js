export const tileData = [
  'color-coral',
  'color-ultramarine',
  'color-deepblue',
  'color-mustard',
  'color-ash',
  'color-peach',
  'color-wine',
  'color-moss',
  'color-mint'
];

export const YahooWeatherUrl = `https://query.yahooapis.com/v1/public/yql?q=select * from weather.forecast where woeid in (select woeid from geo.places(1) where text='new york, ny')&format=json&env=store%3A%2F%2Fdatatables.org%2Falltableswithkeys`;
