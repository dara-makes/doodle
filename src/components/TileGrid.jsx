import React, { Component } from 'react';
import { tileData } from '../data/constants';
import Tile from './Tile';
import './TileGrid.scss';

/**
 * Nav Menu - Collection of color tiles
 *
 * @param
 * @public
 */
class TileGrid extends Component {
  state = {
    tileData: tileData
  };

  render() {
    return (
      <div className="tile-grid">
        {this.state.tileData.map((color, index) => {
          return <Tile key={index} colorMapping={color} onClick={this.props.onClick} />;
        })}
      </div>
    );
  }
}

export default TileGrid;
