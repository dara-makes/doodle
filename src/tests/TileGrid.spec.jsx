import React from 'react';
import { shallow } from 'enzyme';
import TileGrid from '../components/TileGrid';
import { tileData } from '../data/constants';

describe('TileGrid', () => {
  it('should render tile grid', () => {
    const cardComponent = shallow(<TileGrid />);
    expect(cardComponent).toMatchSnapshot();
  });

  it('should render the correct number of tiles', () => {
    const cardComponent = shallow(<TileGrid {...tileData} />);
    expect(cardComponent.find('Tile').length).toEqual(tileData.length);
  });
});
