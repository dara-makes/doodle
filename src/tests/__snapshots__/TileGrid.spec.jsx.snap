// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TileGrid should render tile grid 1`] = `
<div
  className="tile-grid"
>
  <Tile
    colorMapping="color-coral"
    key="0"
  />
  <Tile
    colorMapping="color-ultramarine"
    key="1"
  />
  <Tile
    colorMapping="color-deepblue"
    key="2"
  />
  <Tile
    colorMapping="color-mustard"
    key="3"
  />
  <Tile
    colorMapping="color-ash"
    key="4"
  />
  <Tile
    colorMapping="color-peach"
    key="5"
  />
  <Tile
    colorMapping="color-wine"
    key="6"
  />
  <Tile
    colorMapping="color-moss"
    key="7"
  />
  <Tile
    colorMapping="color-mint"
    key="8"
  />
</div>
`;
